src/components/molecules/PosterBookmark/PosterBookmark.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { MouseEventHandler, useMemo, useCallback } from "react";
+
+import { PosterPageVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+
+import {
+  addPosterToProfile,
+  removePosterFromProfile,
+  addEventToPersonalizedSchedule,
+  removeEventFromPersonalizedSchedule,
+} from "api/profile";
+
+import { Bookmark } from "components/atoms/Bookmark";
+
+import { useUser } from "hooks/useUser";
+import { useVenueEvents } from "hooks/events";
+
+export interface PosterPreviewProps {
+  posterVenue: WithId<PosterPageVenue>;
+}
+
+export const PosterBookmark: React.FC<PosterPreviewProps> = ({
+  posterVenue,
+}) => {
+  const venueId = posterVenue.id;
+
+  const { userWithId } = useUser();
+  const userId = userWithId?.id;
+
+  const venueIds = useMemo(() => [venueId], [venueId]);
+
+  const { events: venueEvents } = useVenueEvents({ venueIds: venueIds });
+
+  const savedPosters = useMemo(() => userWithId?.savedPosters ?? {}, [
+    userWithId,
+  ]);
+
+  const isSaved = useMemo(
+    () => savedPosters[venueId]?.includes(venueId) ?? false,
+    [savedPosters, venueId]
+  );
+
+  const removeBookmarks = useCallback(() => {
+    if (!userId) return;
+    removePosterFromProfile({ venueId, userId });
+    venueEvents.forEach((event) => {
+      removeEventFromPersonalizedSchedule({ event, userId });
+    });
```

Comment:
I think we can refactor `removeEventFromPersonalizedSchedule` and `addEventToPersonalizedSchedule` to make them take not `event`, but `events array` or `event IDs` as a parameter. As both these methods use `updateUserCollection` which accepts the array.
Thus, we'll reduce the number of database update requests.